src/generate-presence.ts

```diff
@@ -0,0 +1,258 @@
+import {OptionalLogger} from '@rocicorp/logger';
+import {
+  IDFromEntityFunc,
+  KeyFromEntityFunc,
+  KeyFromLookupIDFunc,
+  KeyToIDFunc,
+  ListOptionsWithLookupID,
+  Parse,
+  ParseInternal,
+  ReadTransaction,
+  Update,
+  WriteTransaction,
+  deleteImpl,
+  getImpl,
+  hasImpl,
+  initImpl,
+  listEntriesImpl,
+  listIDsImpl,
+  listImpl,
+  maybeParse,
+  mustGetImpl,
+  setImpl,
+  updateImpl,
+} from './generate.js';
+
+export type PresenceEntity = {
+  clientID: string;
+  id: string;
+};
+
+export type OptionalIDs<T extends PresenceEntity> = Partial<T> &
+  Omit<T, keyof PresenceEntity>;
+
+export type LookupID = Partial<PresenceEntity>;
```

Comment:
I might call this `PresenceID` just to make it a little more clear from name only, when using the type from outside the package.

Reply:
Good idea

Reply:
Done in https://github.com/rocicorp/rails/pull/23